### PR TITLE
refactor: use Fail template tag in ERTP and zoe

### PR DESCRIPTION
**Pull request by erights**
Replaces #6622 

Only the ERTP and zoe portions of #6482 

See https://github.com/endojs/endo/pull/1334

### CODE REVIEW

File: packages/ERTP/src/amountMath.js [hunk 1]
```diff
@@ -127,11 +127,10 @@ export const assertValueGetHelpers = value =>
 const optionalBrandCheck = (allegedBrand, brand) => {
   if (brand !== undefined) {
     assertRemotable(brand, 'brand');
-    assert.equal(
-      allegedBrand,
-      brand,
-      X`amount's brand ${allegedBrand} did not match expected brand ${brand}`,
-    );
+    allegedBrand === brand ||
+      Fail`amount's brand ${q(allegedBrand)} did not match expected brand ${q(
```
Comment (erights) [inline packages/ERTP/src/amountMath.js:131]: Note change in semantics LOOSENING security: 

The old code was redacting brands. Something as static as a brand I believe does not need to be redacted.

File: packages/zoe/src/contractSupport/ratio.js [hunk 2]
```diff
@@ -74,9 +68,7 @@ export const makeRatio = (
   denominatorBrand = numeratorBrand,
 ) => {
   denominator > 0n ||
-    assert.fail(
-      X`No infinite ratios! Denominator was 0/${q(denominatorBrand)}`,
-    );
+    Fail`No infinite ratios! Denominator was 0/${q(denominatorBrand)}`;
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractSupport/ratio.js]: I don't like **/** as a separator here. Would you mind changing it to a space?
```suggestion
    Fail`No infinite ratios! Denominator was 0 ${q(denominatorBrand)}`;
```

**Review by Chris-Hibbert — COMMENTED** [review]
LGTM.   One minor cleanup of a (pre-existing) error message.

Hmm. I won't approve yet, since I see more changes coming in....

**Review by Chris-Hibbert — APPROVED** [review]
Looks good with the one previous request.

**Pull request merged**